source/client.c

```diff
@@ -1651,6 +1651,8 @@ static enum aws_mqtt_client_request_state s_subscribe_send(uint16_t packet_id, b
 
     /* This is not necessarily a fatal error; if the subscribe fails, it'll just retry. Still need to clean up though.
      */
+    /* TODO: NOT really agree on retry for this failure, it's reasonable to retry if the error is socket_closed or not
+     * connected, otherwiseit's probably a program error. */
```

Comment:
At least, if the socket is closed/not connected, it's fine to put it as on_going, which will be moved to the pending list when the channel finishes shutdown process. Otherwise, it's should be more complicated to deal with.

Reply:
the behavior here is actually ok, but explain that aws_channel_slot_send_message() can only fail if the channel is already in the middle of shutting down